test/e2e/dhcp_overrides_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/base64"
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25/mo"
+	"gopkg.in/yaml.v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/cluster-api/util"
+
+	types "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+)
+
+type GuestInfoMetadata struct {
+	Network Network `yaml:"network"`
+}
+
+type Network struct {
+	Ethernets map[string]Ethernet `yaml:"ethernets"`
+}
+
+type Ethernet struct {
+	DHCP4Overrides DHCPOverrides `yaml:"dhcp4-overrides"`
+}
+
+type DHCPOverrides struct {
+	SendHostname *bool `yaml:"send-hostname"`
+}
+
+var _ = Describe("DHCPOverrides configuration test", func() {
```

Comment:
This is missing an AfterEach block for cleanup of the cluster. Checkout https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/8cbf21df6b1632b48cf81cdc732bf1e44dca51b8/test/e2e/anti_affinity_test.go#L39-L59